src/test/java/com/fasterxml/jackson/databind/ser/jdk/NumberSerTest.java

```diff
@@ -222,6 +228,24 @@ public void testCustomSerializationBigDecimalAsNumber() throws Exception {
         assertEquals(a2q("{'value':2.0}"), mapper.writeValueAsString(new BigDecimalHolder("2")));
     }
 
+    @Test
+    public void testConfigOverrideJdkNumber() throws Exception {
+        ObjectMapper mapper = jsonMapperBuilder().withConfigOverride(BigDecimal.class,
+                        c -> c.setFormat(JsonFormat.Value.forShape(JsonFormat.Shape.STRING)))
+                .build();
+        String value = mapper.writeValueAsString(new BigDecimal("123.456"));
+        assertEquals(a2q("'123.456'"), value);
+    }
+
+    @Test
+    public void testConfigOverrideNonJdkNumber() throws Exception {
+        ObjectMapper mapper = jsonMapperBuilder().withConfigOverride(MyBigDecimal.class,
```

Comment:
Arguably should work with override for `BigDecimal`, not impl type. But probably depends on exactly how Jackson registers relevant serializer (its `handledType()`) so becomes sort of implementation detail.

But if this is how things currently work, we'll go with that.